### ISSUE TITLE
spedup fetch result from the workers using .get()

### ISSUE THREAD

**Issue by ydave** [celery/celery#4327]
## Checklist

- [ *] I have included the output of ``celery -A proj report`` in the issue.
      4.1.0
- [ *] I have verified that the issue exists against the `master` branch of Celery. 
      Yes

## Steps to reproduce
I have 100K files which I process through workers and generate 3 different JSON strings in return (approx. 250 workers). 

## Expected behavior
Fetching result with .get() method within milliseconds

## Actual behavior
Though the task is dispatched to the workers very quickly (in matter of few mins), It takes approx. 1 sec to read the result using .get() method. With that it will take 100K seconds to fetch all the results and then write to disk. What will be the best way to fetch the results from the workers at same/similar speed like the task is dispatched to them?

**Comment by linar-jether** [comment]
See https://github.com/celery/celery/issues/4331

TL;DR
Use:
```python
# Get all task results in a single request using ResultSet
res = celery.result.ResultSet(async_results)
res.get()
```
